File: Exemplos SmartPOS/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_SmartPOS_Ionic-font/src/intentDigitalHubService/TERMICA/Commands/ImpressaoQRCode.ts
import {TermicaCommand} from '../TermicaCommand';

export class ImpressaoQRCode extends TermicaCommand {
  readonly dados: string;
  readonly tamanho: number;
  readonly nivelCorrecao: number;

  constructor(dados: string, tamanho: number, nivelCorrecao: number) {
    super('ImpressaoQRCode');
    this.dados = dados;
    this.tamanho = tamanho;
    this.nivelCorrecao = nivelCorrecao;
  }

  functionParameters(): object {
    return {
      dados: this.dados,
      tamanho: this.tamanho,
      nivelCorrecao: this.nivelCorrecao,
    };
  }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_SmartPOS_Flutter-font/lib/IntentDigitalHubService/TERMICA/Commands/impressao_qr_code.dart
import '../termica_command.dart';

class ImpressaoQRCode extends TermicaCommand {
  final String _dados;
  final int _tamanho;
  final int _nivelCorrecao;

  ImpressaoQRCode(this._dados, this._tamanho, this._nivelCorrecao)
      : super('ImpressaoQRCode');

  @override
  get functionParametersJson => {
        'dados': _dados,
        'tamanho': _tamanho,
        'nivelCorrecao': _nivelCorrecao,
      };
}
